usb/esp_tinyusb/include/tusb_msc_storage.h

```diff
@@ -0,0 +1,151 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stddef.h>
+#include "esp_err.h"
+#include "esp_partition.h"
+#if SOC_SDMMC_HOST_SUPPORTED
+#include "driver/sdmmc_host.h"
+#endif
+
+
+#if SOC_SDMMC_HOST_SUPPORTED
+/**
+ * @brief Configuration structure for sdmmc initialization
+ *
+ * User configurable parameters that are used while
+ * initializing the sdmmc media.
+ */
+typedef struct {
+    sdmmc_host_t host;
+    sdmmc_slot_config_t slot_config;
+} tinyusb_msc_sdmmc_config_t;
+#endif
+
+/**
+ * @brief Configuration structure for spiflash initialization
+ *
+ * User configurable parameters that are used while
+ * initializing the SPI Flash media.
+ */
+typedef struct {
+    esp_partition_t *data_partition; /* If it is NULL, first partition is used */
+} tinyusb_msc_spiflash_config_t;
+
+/**
+ * @brief Configuration structure for storage initialization
+ */
+typedef struct {
+    union {
+        tinyusb_msc_spiflash_config_t spiflash_config;
+#if SOC_SDMMC_HOST_SUPPORTED
+        tinyusb_msc_sdmmc_config_t sdmmc_config;
+#endif
+    };
+} tinyusb_msc_storage_config_t;
+
+/**
+ * @brief Register storage type spiflash with tinyusb driver
+ *
+ * @return esp_err_t
+ *       - ESP_OK, if success;
+ *       - ESP_ERR_NO_MEM, if there was no memory to allocate storage components;
+ */
+esp_err_t tinyusb_msc_storage_register_driver_spiflash(void);
+
+#if SOC_SDMMC_HOST_SUPPORTED
+/**
+ * @brief Register storage type sd-card with tinyusb driver
+ *
+ * @return esp_err_t
+ *       - ESP_OK, if success;
+ *       - ESP_ERR_NO_MEM, if there was no memory to allocate storage components;
+ */
+esp_err_t tinyusb_msc_storage_register_driver_sdmmc(void);
+#endif
+/**
+ * @brief Deregister storage with tinyusb driver and frees the memory
+ *
+ */
+void tinyusb_msc_storage_deregister_driver(void);
+
+/**
+ * @brief Initialize the Storage.
+ *
+ * Initialize the instance of storage media.
+ * Once the storage is initialized, other storage functions can be used.
+ *
+ * @param config  pointer to tinyusb_msc_storage_config_t to pass
+ * user defined configuration during initialization.
+ * @return esp_err_t
+ *       - ESP_OK, if success;
+ *       - ESP_ERR_NO_MEM, if there was no memory to allocate storage components;
+ */
+esp_err_t tinyusb_msc_storage_init(const tinyusb_msc_storage_config_t *config);
+
+/**
+ * @brief Mount the storage partition locally on the firmware application.
+ *
+ * Get the available drive number. Register spi flash partition.
+ * Connect POSIX and C standard library IO function with FATFS.
+ * Mounts the partition.
+ * This API is used by the firmware application. If the storage partition is
+ * mounted by this API, host (PC) can't access the storage via MSC.
+ *
+ * @param base_path  path prefix where FATFS should be registered
```

Comment:
Given that you have this argument, perhaps `CONFIG_TINYUSB_MSC_MOUNT_PATH` option can be removed? (Just one less way to do the same thing => might lead to a simpler API.)

Reply:
@igrr 

This is done for taking care of the following scenario:
1. Media is exposed over usb host.
2. Next, it is manually ejected.
3. Expected behaviour - media should no longer be exposed over usb host + media should be mounted automatically to MOUNT_PATH.

Current behaviour is point 3 such that it is automatically mounted to CONFIG_TINYUSB_MSC_MOUNT_PATH by component (idf-extra-component) without any intervention of application code. This CONFIG option is to tell the component about the mount path when component has to mount the media by itself. In this case, application code is not calling mount API `esp_err_t tinyusb_msc_storage_mount(const char *base_path);` to mount the storage. I can mention about this default path in doxygen documentation of API tinyusb_msc_storage_mount().